hack/deploy/configure-values.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script interrogates the AKS cluster and Azure resources to generate
+# the karpenter-values.yaml file using the karpenter-values-template.yaml file as a template.
+
+# Check the cluster name and resource group are provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <cluster-name> <resource-group>"
+    exit 1
+fi
+
+echo "Configuring karpenter-values.yaml for cluster $1 in resource group $2 ..."
+
+CLUSTER_NAME=$1
+AZURE_RESOURCE_GROUP=$2
+
+AKS_JSON=$(az aks show --name "$CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP")
+AZURE_LOCATION=$(jq -r ".location" <<< "$AKS_JSON")
+AZURE_RESOURCE_GROUP_MC=$(jq -r ".nodeResourceGroup" <<< "$AKS_JSON")
+
+KARPENTER_SERVICE_ACCOUNT_NAME=karpenter-sa
+AZURE_KARPENTER_USER_ASSIGNED_IDENTITY_NAME=karpentermsi
```

Comment:
I'd like to reuse these variables if possible. Could we pass them in as params?

nit-concern on having them defined in two places for future updates.

Reply:
Fixed